fibers/web/server.scm

```diff
@@ -124,13 +115,19 @@ on the procedure being called at any particular time."
     (error "response with this status code must not include body" response))
    (else
     ;; check length; assert type; add other required fields?
-    (values (let ((rlen (response-content-length response))
-                  (blen (bytevector-length body)))
-              (cond
-               (rlen (if (= rlen blen)
-                         response
-                         (error "bad content-length" rlen blen)))
-               (else (extend-response response 'content-length blen))))
+    (values (if (procedure? body)
+                (if (response-content-length response)
+                    response
+                    (extend-response response
+                                     'transfer-encoding
+                                     '((chunked))))
+                (let ((rlen (response-content-length response))
+                      (blen (bytevector-length body)))
+                  (cond
+                   (rlen (if (= rlen blen)
+                             response
+                             (error "bad content-length" rlen blen)))
```

Comment:
Is this an error that can be caught (and not hidden in some new fiber or something)?  If so, a separate condition type can be useful for more convenient catching.

Reply:
I'm just moving this code around, so I've avoided changing it.